src/deluge/gui/views/instrument_clip_view.cpp

```diff
@@ -1302,69 +1332,71 @@ ActionResult InstrumentClipView::padAction(int32_t x, int32_t y, int32_t velocit
 				if (afh == NULL) {
 					continue;
 				}
-
-				static int32_t MaxFiles = 25;
-				String fnArray[MaxFiles];
-				char const* currentPathChars = afh->filePath.get();
-				if (currentPathChars == NULL) {
+				char const* path = afh->filePath.get();
+				if (path == NULL) {
 					continue;
 				}
-				char const* slashAddress = strrchr(currentPathChars, '/');
-				if (slashAddress) {
-					int32_t slashPos = (uint32_t)slashAddress - (uint32_t)currentPathChars;
-					String dir;
-					dir.set(&afh->filePath);
-					dir.shorten(slashPos);
-					FRESULT result = f_opendir(&staticDIR, dir.get());
-					FilePointer thisFilePointer;
-					int32_t numSamples = 0;
-
-					if (result != FR_OK) {
-						display->displayError(ERROR_SD_CARD);
-						return ActionResult::DEALT_WITH;
+				char* slashAddress = strrchr(path, '/');
+				if (slashAddress == NULL) {
+					continue;
+				}
+
+				// Open directory of current audio file
+				*slashAddress = 0;
+				FRESULT result = f_opendir(&staticDIR, path);
+				if (result != FR_OK) {
+					display->displayError(ERROR_SD_CARD);
+					return ActionResult::DEALT_WITH;
```

Comment:
I think this will leave the `AudioFileHolder` in a bad state, since `slashAddress` writes directly to the underlying memory. Instead, we should copy the directory name in to `chosenFilename`.

Reply:
Done!

Oops, I assumed that `get()` retuned a copy!

I opted for just writing the slash back into `slashAddress` right after using the shortened path to open the dir. 
As paths can theoretically be up to 65k chars long, I want to avoid copying or making `chosenFilename` that big.

Admittedly, this leaves the "bad state" for the duration of the `f_opendir` call, but I think the afh thrying to reload the file at that exact instant is, if not plain impossible, *extremely* unlikely...